src/interpreter.rs

```diff
@@ -120,3 +129,1207 @@ bitflags::bitflags! {
+                                vch_hash = Sha256::digest(Sha256::digest(vch)).to_vec();
+                            }
+                            stack.pop()?;
+                            stack.push_back(vch_hash)
+                        }
+
+                        OP_CHECKSIG
+                        | OP_CHECKSIGVERIFY => {
+                            // (sig pubkey -- bool)
+                            if stack.size() < 2 {
+                                return set_error(ScriptError::InvalidStackOperation);
+                            }
+
+                            let vch_sig = stack.top(-2)?.clone();
+                            let vch_pub_key = stack.top(-1)?.clone();
+
+                            if !check_signature_encoding(&vch_sig, flags)? {
+                                //serror is set
+                                return Ok(false);
+                            }
+                            check_pub_key_encoding(&vch_pub_key, flags)?;
+                            let success = checker.check_sig(&vch_sig, &vch_pub_key, script);
+
+                            stack.pop()?;
+                            stack.pop()?;
+                            stack.push_back(if success {
+                                vch_true.clone()
+                            } else {
+                                vch_false.clone()
+                            });
+                            if op == OP_CHECKSIGVERIFY {
+                                if success {
+                                    stack.pop()?;
+                                } else {
+                                    return set_error(ScriptError::CheckSigVerify);
+                                }
+                            }
+                        }
+
+                        OP_CHECKMULTISIG
+                        | OP_CHECKMULTISIGVERIFY => {
+                            // ([sig ...] num_of_signatures [pubkey ...] num_of_pubkeys -- bool)
+
+                            // NB: This is guaranteed u8-safe, because we are limited to 20 keys and
+                            //     20 signatures, plus a couple other fields. u8 also gives us total
+                            //     conversions to the other types we deal with here (`isize` and `i64`).
+                            let mut i: u8 = 1;
+                            if stack.size() < i.into() {
+                                return set_error(ScriptError::InvalidStackOperation);
+                            };
+
+                            let mut keys_count =
+                                u8::try_from(ScriptNum::new(stack.top(-isize::from(i))?, require_minimal, None)?)
+                                  .map_err(|_| ScriptError::PubKeyCount)?;
+                            if keys_count > 20 {
+                                return set_error(ScriptError::PubKeyCount);
+                            };
+                            op_count += keys_count;
```

Comment:
On nightly we have `strict_add` which I'd suggest here. It's a little too verbose with `.checked_add` and `.expect`.